internal/rbd/rbd_journal.go

```diff
@@ -287,6 +283,10 @@ func (rv *rbdVolume) Exists(ctx context.Context, parentVol *rbdVolume) (bool, er
 	if rv.Topology != nil {
 		rv.Pool = imageData.ImagePool
 	}
+	// validate rbdvolume pool only after looking to the topology pools
```

Comment:
it was added above so removed from there,
No the `pool` will be updated with `topologypool` so  checking it

Reply:
@parth-gr, instead of moving the `validateNonEmptyField` here. 
Why not call the `validateRbdVol()` function?

https://github.com/ceph/ceph-csi/blob/991343d9e5ee5780317df6976e206a864e6b94c0/internal/rbd/rbd_journal.go#L254-L257

Reply:
i still see no reason to move this check here, please move this check to validateRbdVol

Reply:
We are moving it here, as if you see line above it
```
if rv.Topology != nil {
		rv.Pool = imageData.ImagePool
	}
```
We are updating the pool there,

And if we check in the start of the function at `validateRbdVol` it fail as it would be empty,

So only check when we make sure it will be non empty

Reply:
No need to validation in this function as its retrieved from the omap, we don't do any validation in the `Exists` function, validation need to be done at higher level like `validateRbdVol` function